### ISSUE TITLE
Uninitialized constant SCSSLint::LinterError when passing configs into SCSSLint::RakeTask

### ISSUE THREAD

**Issue by SebAshton** [causes/scss-lint#353]
0.33.0 works fine when I use `scss-lint app/assets/stylesheets` however when trying to use the new rake task I get the following error.

**scss_lint.rake**
```ruby
  require 'scss_lint/rake_task'

  desc 'Run scss lint against SCSS files'
  SCSSLint::RakeTask.new do |t|
    t.config = '.scss_lint.yml'
    t.files = ['app/assets/stylesheets']
  end
```

**output**
```shell
SCSSLint::Linter::PropertySortOrder raised unexpected error linting file app/assets/stylesheets/_bootstrap.scss: 'uninitialized constant SCSSLint::LinterError'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/linter/property_sort_order.rb:103:in `rescue in extract_preferred_order_from_config'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/linter/property_sort_order.rb:97:in `extract_preferred_order_from_config'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/linter/property_sort_order.rb:7:in `visit_root'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/sass-3.4.10/lib/sass/tree/visitors/base.rb:36:in `visit'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/linter.rb:127:in `visit'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/linter.rb:21:in `run'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/runner.rb:58:in `run_linter'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/runner.rb:39:in `block in find_lints'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/runner.rb:37:in `each'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/runner.rb:37:in `find_lints'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/runner.rb:21:in `block in run'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/runner.rb:20:in `each'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/runner.rb:20:in `run'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/cli.rb:52:in `scan_for_lints'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/cli.rb:46:in `act_on_options'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/cli.rb:25:in `run'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/rake_task.rb:70:in `run_cli'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/scss-lint-0.33.0/lib/scss_lint/rake_task.rb:63:in `block in define'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/usr/local/rbenv/versions/2.1.2/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:150:in `invoke_task'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `each'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `block in top_level'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:115:in `run_with_threads'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:100:in `top_level'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:78:in `block in run'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/rake-10.4.2/bin/rake:33:in `<top (required)>'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/bin/rake:23:in `load'
/usr/local/rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/bin/rake:23:in `<main>'
```

**Comment by sds** [comment]
I'm not sure why you're seeing a difference between running `scss-lint` normally and via the rake task, but the error you're seeing is definitely a bug that would happen regardless.

I've fixed that bug in 1a12417d69a390748825a07ea124fbe7dd078076. Feel free to file another bug if there is a separate problem at play here.

**Comment by SebAshton** [comment]
@sds I updated my bundle to that ref and the issue still persisted.

However I tracked the issue do passing `order: nil` into the the `PropertySortOrder` linter. I removed this and the rake task worked perfectly.

**Comment by sds** [comment]
@SebAshton yes, this is expected behavior. 

Unfortunately, passing the string `nil` in YAML does not actually set the value to `nil`, but rather to the string `"nil"`. The `~` character is reserved for setting a value to `nil` in YAML (not obvious, I know).

Thus the best thing to do is to not specify `order` at all in this case, as you have done. Hope that explains things!
